### ISSUE TITLE
getGlobal

### ISSUE THREAD

**Issue by lishine** [CharlesStover/reactn#35]

**Comment by CharlesStover** [comment]
It isn't currently publicly accessible, or if it is it's not by design. It follows the law of encapsulation. All exposed states are copies of the state such that direct access to mutate isn't possible.

What is your use case for copying state outside of React, and maybe it can be worked around, or maybe it will be convincing to publicly expose the global state?

**Comment by lishine** [comment]
I just mean to use it like store.setState({count:1}) outside of react component. Like with easy-peasy

**Comment by CharlesStover** [comment]
_Setting_ the state can, yes. It's not recommended over `this.setGlobal` or the `useGlobal` hook, but there are no current or expected side effects of doing so.

```JavaScript
import { setGlobal } from 'reactn';
```

Let me know if you have any further questions or if this issue can sufficiently be closed.

**Comment by umbertoghio** [comment]
Hi, 
I am trying out ReactN on a new project, so far so good :-) 

As I am working on a library function it will be glad to access the store directly (now I am passing this.global from reactn components).

I am now trying to import setGlobal directly, but it will be great to have a way to access "this.global" outside of a reactn component... as workaround I also found that I can make a class that extends react(n).component but I like more regular functions

Thanks!

**Comment by umbertoghio** [comment]
that would be great! 

I switched one project to Reactn and it’s incredible how easy it is to manage global state....

setGlobal() works perfectly inside library functions, while creating a class component 
without render (just to access state) is not an option as it get hit with a force update when setting global state. 

Thank you very much for your great work!

**Comment by CharlesStover** [comment]
I'll try to have an iteration of this soon for single-store support only.

**Comment by CharlesStover** [comment]
I'd just like to state that I have this existing in my local repository, but it's pretty important to me that I get unit tests completed for the project as a whole before cutting a new version, so I do not have an ETA on release. I may do an alpha release if the unit tests delay this too much.

**Comment by umbertoghio** [comment]
I am currently migrating a medium size (25 screens) Expo React Native project from Unstated to Reactn, if you release the alpha version with GetGlobal I can help with testing it out :-)
Thank you for your work!

**Comment by CharlesStover** [comment]
I have released `reactn@0.2.0` to include this functionality as `getGlobal`. The documentation for `getGlobal` has been [added to the README](https://github.com/CharlesStover/reactn#getglobal).

I added some unit tests and manual app testing to try to insure there were no breaking changes, but this version included a major change to some internal workings. I would love some community feedback on if any errors are discovered.

Thank you for your patience. I'm sorry about the long delay. I have been unable to contribute for the past month, but I'm hoping to begin knocking out a lot of ReactN's open issues now.
